apache/superset
# refactor(dashboard): Migrate ResizableContainer to TypeScript and functional component

### SUMMARY


- Refactoring the component to use React functional component syntax
- Defining TypeScript interfaces for props and state
- Adding type safety throughout the component
- Performance improving
### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

- Before 
![Screenshot 2025-01-09 143325](https://github.com/user-attachments/assets/86f41ab8-6ddf-4e60-9f4f-d6ac0a934425)

https://github.com/user-attachments/assets/d41e4b07-3882-4a51-8ef3-5ee6da0a006b

- After
![Screenshot 2025-01-09 115936](https://github.com/user-attachments/assets/f4ab2e7d-c40d-4d0d-9943-79b85cc92fa3)

https://github.com/user-attachments/assets/d375acb5-355a-45b0-b39c-102e7e861a05


### TESTING INSTRUCTIONS


1. Navigate to superset-frontend/src/dashboard/components/resizable.
2. Run the command: npm run test ResizableContainer.test.tsx.
3. Verify that all tests pass successfully.


### ADDITIONAL INFORMATION


- [x ] Changes UI
